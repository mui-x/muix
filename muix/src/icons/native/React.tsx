import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)