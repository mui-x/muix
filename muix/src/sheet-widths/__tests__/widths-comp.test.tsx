/** @jsx platform.createElement */
import React from 'react'
import { platform } from "reactxx-sheeter";

import { ts, traceComponentEx, ReactAny, setActWidth } from "reactxx-tests";
import { sheetWidths_registerVariantHandler, WidthsProvider, widthsPipe, onWidthChanged } from "../index";
import { Provider } from 'react-fela';

sheetWidths_registerVariantHandler()

describe("WIDTHS COMPONENT", () => {
  it("NATIVE 300", () => traceTest(false, 300))
  it("NATIVE 800", () => traceTest(false, 800))
  it("NATIVE 1200", () => traceTest(false, 1200))
  it("WEB 300", () => traceTest(true, 300))
  it("WEB 800", () => traceTest(true, 800))
  it("WEB 1200", () => traceTest(true, 1200))
})

const traceTest = (isWeb: boolean, actWidth: number) => {

  traceComponentEx(isWeb, { traceLevel: 2, actWidth },

    {
      root: ts.view = {
        $widths: {
          '-640': {
            backgroundColor: 'blue'
          },
          '640-1024': {
            backgroundColor: 'green'
          },
          '1024-': {
            backgroundColor: 'red'
          },
        }
      },
      label: {}, webOnly: {}, nativeOnly: {},
    },

    ({ classes, toClassNames, actWidths }) => {
      return <ReactAny classNameX={toClassNames([classes.root])} actWidths={actWidths} actWidth={platform.actWidth()}>
        {actWidths.mobileWidth ? 'MOBILE' : actWidths.tabletWidth ? 'TABLET' : 'DESKTOP'}
      </ReactAny>
    },

    Comp => <WidthsProvider><Comp /></WidthsProvider>,

    {
      nativeHasWidthRule: true,
      getPipes: options => ({
        afterPropsCode: [widthsPipe],
      }),
      defaultProps: {
        actWidths: {
          mobileWidth: '-640',
          tabletWidth: '640-1024',
          desktopWidth: '1024-'
        }
      }
    },

    (wrapper, Comp) => {
      expect(wrapper).toMatchSnapshot()

      setActWidth(platform.actWidth() + 444)
      onWidthChanged()
      wrapper.update()

      expect(wrapper).toMatchSnapshot();
    }
  )
}
      // const myComponents = wrapper.find(Comp)
      // const props = myComponents.instance()
