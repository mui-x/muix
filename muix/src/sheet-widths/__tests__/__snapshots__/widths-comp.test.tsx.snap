// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WIDTHS COMPONENT NATIVE 300 1`] = `
<WidthsProvider>
  <ReactAny
    actWidths={
      Object {
        "mobileWidth": true,
      }
    }
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
    trace={
      Object {
        "backgroundColor": "blue /*TestComponent.sheet.root/$widths.-640*/",
      }
    }
  >
    MOBILE
  </ReactAny>
</WidthsProvider>
`;

exports[`WIDTHS COMPONENT NATIVE 800 1`] = `
<WidthsProvider>
  <ReactAny
    actWidths={
      Object {
        "tabletWidth": true,
      }
    }
    style={
      Object {
        "backgroundColor": "green",
      }
    }
    trace={
      Object {
        "backgroundColor": "green /*TestComponent.sheet.root/$widths.640-1024*/",
      }
    }
  >
    TABLET
  </ReactAny>
</WidthsProvider>
`;

exports[`WIDTHS COMPONENT NATIVE 1200 1`] = `
<WidthsProvider>
  <ReactAny
    actWidths={
      Object {
        "desktopWidth": true,
      }
    }
    style={
      Object {
        "backgroundColor": "red",
      }
    }
    trace={
      Object {
        "backgroundColor": "red /*TestComponent.sheet.root/$widths.1024-*/",
      }
    }
  >
    DESKTOP
  </ReactAny>
</WidthsProvider>
`;

exports[`WIDTHS COMPONENT WEB 300 1`] = `
<WidthsProvider>
  <ReactAny
    actWidths={
      Object {
        "mobileWidth": true,
      }
    }
    className="c b a"
    trace={
      Array [
        "@media (min-width: 1024px).c { background-color:red /*TestComponent.sheet.root/$widths.1024-*/ }",
        "@media (min-width: 640px) and (max-width: 1023px).b { background-color:green /*TestComponent.sheet.root/$widths.640-1024*/ }",
        "@media (max-width: 639px).a { background-color:blue /*TestComponent.sheet.root/$widths.-640*/ }",
      ]
    }
  >
    MOBILE
  </ReactAny>
</WidthsProvider>
`;

exports[`WIDTHS COMPONENT WEB 800 1`] = `
<WidthsProvider>
  <ReactAny
    actWidths={
      Object {
        "tabletWidth": true,
      }
    }
    className="c b a"
    trace={
      Array [
        "@media (min-width: 1024px).c { background-color:red /*TestComponent.sheet.root/$widths.1024-*/ }",
        "@media (min-width: 640px) and (max-width: 1023px).b { background-color:green /*TestComponent.sheet.root/$widths.640-1024*/ }",
        "@media (max-width: 639px).a { background-color:blue /*TestComponent.sheet.root/$widths.-640*/ }",
      ]
    }
  >
    TABLET
  </ReactAny>
</WidthsProvider>
`;

exports[`WIDTHS COMPONENT WEB 1200 1`] = `
<WidthsProvider>
  <ReactAny
    actWidths={
      Object {
        "desktopWidth": true,
      }
    }
    className="c b a"
    trace={
      Array [
        "@media (min-width: 1024px).c { background-color:red /*TestComponent.sheet.root/$widths.1024-*/ }",
        "@media (min-width: 640px) and (max-width: 1023px).b { background-color:green /*TestComponent.sheet.root/$widths.640-1024*/ }",
        "@media (max-width: 639px).a { background-color:blue /*TestComponent.sheet.root/$widths.-640*/ }",
      ]
    }
  >
    DESKTOP
  </ReactAny>
</WidthsProvider>
`;
